src/components/ContentList/ContentList.tsx

```diff
@@ -25,25 +25,34 @@ function getHeadingLevel(size: ContentSize) {
     }
 }
 
-interface ContentListProps {
-    list: ContentItemProps[];
-    size: ContentSize;
+function isIconSvg(icon: ContentItemProps['icon']): icon is SVGIcon {
+    if (typeof icon === 'function') {
+        return true;
+    }
+    return false;
 }
 
-const ContentList = ({list, size, qa}: ContentListProps & QAProps) => {
+const ContentList = ({list, size = 'l', qa}: ContentListProps & QAProps) => {
     const theme = useTheme();
     const qaAttributes = getQaAttrubutes(qa, ['image', 'title', 'text']);
 
+    const renderIcon = (icon: ContentItemProps['icon']): JSX.Element => {
+        const iconThemed = getThemedValue(icon, theme);
+        if (isIconSvg(iconThemed)) {
+            const Icon = iconThemed;
+            return <div>{<Icon className={b('icon')} />}</div>;
+        }
+        const iconData = getMediaImage(iconThemed);
+        return <Image {...iconData} className={b('icon')} qa={qaAttributes.image} />;
+    };
+
```

Comment:
can we moved this in a separate  file component ? together with this utility `isIconSvg`

Reply:
It's possible. But it seems like that renderIcon strongly relates to parent component through theme and qaAttributes params. @yuberdysheva What do you think? Is it worth it?